pkg/services/validator/vmware.go

```diff
@@ -526,19 +462,26 @@ func readEntityPrivilegeRule(ctx context.Context, c *components.VspherePluginCon
 	return nil
 }
 
-func readCustomEntityPrivileges(ctx context.Context, c *components.VspherePluginConfig, r *components.VsphereEntityPrivilegeRule, driver vsphere.VsphereDriver) error {
+func readCustomEntityPrivileges(ctx context.Context, c *components.VspherePluginConfig, r *components.VsphereEntityPrivilegeRule, driver vsphere.VsphereDriver, isAdmin bool) error {
 	var err error
 	if r.Username == "" {
-		r.Username, err = prompts.ReadTextRegex("vSphere username to validate entity privileges for", r.Username, "Invalid vSphere username", cfg.VSphereUsernameRegex)
-		if err != nil {
-			return err
+
+		if isAdmin {
+			r.Username, err = prompts.ReadTextRegex("vSphere username to validate entity privileges for", r.Username, "Invalid vSphere username", cfg.VSphereUsernameRegex)
+			if err != nil {
+				return err
+			}
+		} else {
+			log.InfoCLI(`Privilege validation rule will be applied for username %s`, c.Account.Username)
+			r.Username = c.Account.Username
 		}
+
 		r.EntityType, r.EntityName, r.ClusterName, err = getEntityInfo(ctx, "", "Entity Type", c.Validator.Datacenter, cfg.ValidatorPluginVsphereEntities, driver)
 		if err != nil {
 			return err
 		}
 	}
-	privileges, err := LoadPrivileges("vsphere-root-level-permissions-all.yaml")
+	privileges, err := LoadPrivileges("vsphere-root-level-privileges-all.yaml")
```

Comment:
this was a pre-existing bug. custom entity privilege rules never worked because the referenced file never existed.